include/wil/network.h

```diff
@@ -0,0 +1,1338 @@
+        // - so that one can call set_port() and set_address() in any order with expected results
+        // the address family is preserved unless it is specified (or inferred) as an argument
+        void set_address_any() WI_NOEXCEPT;
+        void set_address_any(ADDRESS_FAMILY family) WI_NOEXCEPT;
+        void set_address_loopback() WI_NOEXCEPT;
+        void set_address_loopback(ADDRESS_FAMILY family) WI_NOEXCEPT;
+        void set_address(const IN_ADDR*) WI_NOEXCEPT;
+        void set_address(const IN6_ADDR*) WI_NOEXCEPT;
+
+        void set_port(USHORT) WI_NOEXCEPT;
+        void set_scope_id(ULONG) WI_NOEXCEPT;
+        void set_flow_info(ULONG) WI_NOEXCEPT;
+
+        // format_address prints the IP address, not the scope id or port
+        HRESULT format_address_nothrow(socket_address_wstring& address) const WI_NOEXCEPT;
+        HRESULT format_address_nothrow(socket_address_string& address) const WI_NOEXCEPT;
+
+        // format_complete_address_nothrow() prints the IP address as well as the scope id and port values
+        HRESULT format_complete_address_nothrow(socket_address_wstring& address) const WI_NOEXCEPT;
+        HRESULT format_complete_address_nothrow(socket_address_string& address) const WI_NOEXCEPT;
+
+#if (WIL_USE_STL && defined(WIL_ENABLE_EXCEPTIONS)) || defined(WIL_DOXYGEN)
+        [[nodiscard]] ::std::wstring format_address() const;
+        [[nodiscard]] ::std::wstring format_complete_address() const;
+#endif
+
+        // type: NlatUnspecified ('any'), NlatUnicast, NlatAnycast, NlatMulticast, NlatBroadcast, NlatInvalid
+        [[nodiscard]] NL_ADDRESS_TYPE address_type() const WI_NOEXCEPT;
+        [[nodiscard]] bool is_address_linklocal() const WI_NOEXCEPT;
+        [[nodiscard]] bool is_address_loopback() const WI_NOEXCEPT;
+
+        [[nodiscard]] ADDRESS_FAMILY family() const WI_NOEXCEPT;
+        [[nodiscard]] USHORT port() const WI_NOEXCEPT;
+        [[nodiscard]] ULONG flow_info() const WI_NOEXCEPT;
+        [[nodiscard]] ULONG scope_id() const WI_NOEXCEPT;
+
+        [[nodiscard]] SOCKADDR* sockaddr() WI_NOEXCEPT;
+        [[nodiscard]] SOCKADDR_IN* sockaddr_in() WI_NOEXCEPT;
+        [[nodiscard]] SOCKADDR_IN6* sockaddr_in6() WI_NOEXCEPT;
+        [[nodiscard]] SOCKADDR_INET* sockaddr_inet() WI_NOEXCEPT;
+        [[nodiscard]] IN_ADDR* in_addr() WI_NOEXCEPT;
+        [[nodiscard]] IN6_ADDR* in6_addr() WI_NOEXCEPT;
+
+        [[nodiscard]] const SOCKADDR* sockaddr() const WI_NOEXCEPT;
+        [[nodiscard]] const SOCKADDR_IN* sockaddr_in() const WI_NOEXCEPT;
+        [[nodiscard]] const SOCKADDR_IN6* sockaddr_in6() const WI_NOEXCEPT;
+        [[nodiscard]] const SOCKADDR_INET* sockaddr_inet() const WI_NOEXCEPT;
+        [[nodiscard]] const IN_ADDR* in_addr() const WI_NOEXCEPT;
+        [[nodiscard]] const IN6_ADDR* in6_addr() const WI_NOEXCEPT;
+
+        [[nodiscard]] SOCKADDR_STORAGE sockaddr_storage() const WI_NOEXCEPT;
+
+        [[nodiscard]] constexpr int length() const

```

Comment:
```suggestion
        [[nodiscard]] constexpr int size() const
```
Length implies enumerability (e.g. an array) whereas this is communicating the size of a structure